### DISCUSSION TITLE
Get Started with Apicurio Registry Results in 404

### DISCUSSION THREAD

**Discussion by DarenThomas** [Apicurio/apicurio-registry#5453] [Q&A]
I followed the [Get Started](https://www.apicur.io/registry/getting-started/) instructions, but when I go to http://localhost:8888 I get a `404 page not found` error with version 3.0.3 of the containers.  About a month ago this worked with version 3.0.0 of the containers.  I'm wondering if there is something I can look at in the containers to verify proper behavior of the containers.  I'm seeing the same error with version 3.0.0 now, so I suspect it's my system, but I don't know what it could be.

**Comment by DarenThomas** [comment]
I should mention that http://localhost:8080 is able to access the registry.  It's only the registry-ui that is experiencing the `404` errors.

**Comment by DarenThomas** [comment]
I updated Docker Desktop and now it's working.  Definitely something wrong with my system, although I'm not exactly sure what.  Sorry for the question, but maybe I just needed to ask it in order to fix it.

**Comment by Ganesh-403** [comment]
@DarenThomas Since you mentioned that:

- the Registry API was reachable at `http://localhost:8080`, and
- only the UI at `http://localhost:8888` returned a `404`,

it sounds like the backend container was healthy and the issue was isolated to the UI container or its runtime environment.

The current "Getting Started" guide still expects the UI to be served from a **separate `apicurio-registry-ui` container** on port `8888`, while the Registry API runs on port `8080`. :contentReference[oaicite:0]{index=0}

Since updating Docker Desktop resolved the issue, I don't think this was an Apicurio Registry bug. It was most likely a local Docker/runtime issue that prevented the UI container from serving its static content correctly.

For anyone who encounters similar symptoms, I'd recommend checking:

1. Verify that the UI container is actually running:

   ```bash
   docker ps
   ```

2. Check the UI container logs:

   ```bash
   docker logs <registry-ui-container>
   ```

3. Confirm that the UI container is exposing port `8080` internally and is mapped to `8888` on the host:

   ```bash
   docker port <registry-ui-container>
   ```

4. If the backend is reachable but the UI returns `404`, try pulling the latest UI image and recreating the container:

   ```bash
   docker pull apicurio/apicurio-registry-ui:3.0.3
   docker rm -f <registry-ui-container>
   docker run -it -p 8888:8080 apicurio/apicurio-registry-ui:3.0.3
   ```

5. If you're using Docker Desktop, ensure it's up to date and restart the Docker engine. As you discovered, an outdated or inconsistent Docker installation can sometimes cause unexpected behavior even when the container appears to be running.

I'm glad you found the root cause. Hopefully this information helps anyone else who runs into the same issue.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
